packages/backend/src/user/service.ts

```diff
@@ -0,0 +1,42 @@
+import { Logger as PinoLogger } from '../logger/service'
+import { User } from './model'
+import { TransactionOrKnex } from 'objection'
+
+type Logger = typeof PinoLogger
+
+export interface UserService {
+  get(id: string): Promise<User>
+  create(): Promise<User>
+}
+
+interface ServiceDependencies {
+  logger: Logger
+  knex: TransactionOrKnex
+}
+
+export async function createUserService(
```

Comment:
Why not pass in a (destructured) `ServiceDependencies` instead of individual arguments?

Reply:
1. It's better to be explicit - this helps with knowing what a service requires as inputs (remember this function is used to instantiate the service).
2. You don't need to structure an object when instantiating the service.
3. You often need to change/format some inputs before passing them on to children services. Thus it makes sense to structure that object within the service.